src/tasks/test-execute-proposal.ts

```diff
@@ -0,0 +1,58 @@
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { Args } from "../ts/lib/common-interfaces";
+
+import { generateDeployment, readSettings } from "./ts/proposal";
+
+const OUTPUT_FOLDER = "./output/test-execute-proposal";
+
+const setupTestExecuteProposalTask: () => void = () => {
+  task(
+    "test-execute-proposal",
+    `This script generates a proposal (like the deployment script) but also executes all transactions onchain.`,
+  )
+    .addParam(
+      "claims",
+      "Path to the CSV file that contains the list of claims to generate.",
+    )
+    .addParam(
+      "settings",
+      "Path to the JSON file that contains the deployment settings.",
+    )
+    .setAction(executeProposal);
+};
+
+export { setupTestExecuteProposalTask };
+
+async function executeProposal(
+  args: Args,
+  hre: HardhatRuntimeEnvironment,
+): Promise<void> {
+  const settings = await readSettings(args.settings);
+  const [gnosisDao] = await hre.ethers.getSigners();
+  settings.gnosisDao = gnosisDao.address;
```

Comment:
Another advantage of forking in a test is that we are able to impersonate the Gnosis DAO, so we don't need to do this.